gateway/src/service.rs

```diff
@@ -940,9 +947,34 @@ impl GatewayService {
         &self.auth_host
     }
 
-    /// Maximum number of containers that can be started by gateway
-    pub fn container_limit(&self) -> u32 {
-        self.container_limit
+    /// Is there enough capacity to start this project
+    ///
+    /// There is capacity if we are below the cch limit.
+    /// Else free and pro tier projects below the soft limit is allowed.
+    /// But only pro tier projects are allowed between the soft and hard limit.
+    /// Nothing should be allowed above the hard limits so that our own services don't crash.
+    pub async fn has_capacity(
+        &self,
+        is_cch_project: bool,
+        account_tier: &AccountTier,
+    ) -> Result<(), Error> {
+        let current_container_count = self.count_ready_projects().await?;
+
+        let has_capacity = if current_container_count < self.cch_container_limit {
+            true
+        } else if current_container_count < self.soft_container_limit {
+            !is_cch_project
+        } else if current_container_count < self.hard_container_limit {
+            matches!(account_tier, AccountTier::Pro)
+        } else {
+            false
+        };
```

Comment:
Should we add tracing events with the count if we are over any of the limits?

Reply:
I'm honestly not sure. Because of the ambulance event (in the other PR) we can already setup a honeycomb alert. So I want to add a `warn!` in this function but don't see how it will be useful considering the ambulance event.

Reply:
True, it's likely overkill. 👍